src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,579 @@
+        }
+
+        internal int BaudRate
+        {
+            set
+            {
+                if (value != _baudRate)
+                {
+                    if (value <= 0 || value > 230400)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(BaudRate), SR.ArgumentOutOfRange_NeedPosNum);
+                    }
+
+                    if (Interop.Termios.TermiosSetSpeed(_handle, value) < 0)
+                    {
+                        throw new IOException();
+                    }
+
+                    _baudRate = value;
+                }
+            }
+
+            get
+            {
+                return Interop.Termios.TermiosGetSpeed(_handle);
+            }
+        }
+
+        public bool BreakState
+        {
+            get { return _inBreak; }
+            set
+            {
+                throw new System.PlatformNotSupportedException(SR.PlatformNotSupported_IOPorts);
+            }
+        }
+
+        internal int BytesToWrite
+        {
+            get { return Interop.Termios.TermiosGetAvailableBytes(_handle, Interop.Termios.Queue.SendQueue); }
+        }
+
+        internal int BytesToRead
+        {
+            get { return Interop.Termios.TermiosGetAvailableBytes(_handle, Interop.Termios.Queue.ReceiveQueue); }
+        }
+
+        internal bool CDHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDcd);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool CtsHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalCts);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool DsrHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDsr);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool DtrEnable
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDtr);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+
+            set
+            {
+                if (Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDtr, value ? 1 : 0) != 0)
```

Comment:
I don't remember the native layer but I have to ask: does `TermisGetSignal` actually sets a value? Notice also that this code is also repeated with the exception of the signal.